### ISSUE TITLE
Test suite can not be run in parallel.

### ISSUE THREAD

**Issue by demerphq** [tomhughes/IO-Zlib#5]
The test suite fails when run in parallel:

```
$ prove -b -j4 
t/basic.t ..... ok                                                      
===(      18;0   1/35  0/?  0/?  0/? )==================================gzip: test.gz: No such file or directory
Can't call method "read" on an undefined value at t/external.t line 101.
t/external.t .. Dubious, test returned 2 (wstat 512, 0x200)             
Failed 10/35 subtests 
t/getline.t ... ok                                                      
t/getc.t ...... Failed 1/10 subtests                                    
t/import.t .... ok                                                      
t/large.t ..... ok                                                      
t/tied.t ...... ok                                                      
t/uncomp1.t ... ok                                                      
t/uncomp2.t ... ok     

Test Summary Report
-------------------
t/external.t (Wstat: 512 Tests: 28 Failed: 3)
  Failed tests:  22-24
  Non-zero exit status: 2
  Parse errors: Bad plan.  You planned 35 tests but ran 28.
t/getc.t    (Wstat: 0 Tests: 10 Failed: 1)
  Failed test:  9
Files=9, Tests=117,  0 wallclock secs ( 0.05 usr  0.00 sys +  0.27 cusr  0.02 csys =  0.34 CPU)
Result: FAIL
```

this is because it uses the same file names in all of its tests, creating a race on file creation etc.  I pushed

https://github.com/tomhughes/IO-Zlib/pull/4 

to fix. In perl core we would like to get all our tests working in parallel so we can improve the time it takes to test a new perl. This is the last fix we need to apply to be able to fully test perl core in parallel safely.

**Comment by tomhughes** [comment]
What is the point of opening a duplicate issue after you have already opened a PR for this? I will look at you PR when I have time and in the meantime please stop wasting my time by creating unnecessary extra work!

Closing as duplicate of #4.

**Comment by demerphq** [comment]
FWIW, in perl-core these days we normally create an issue *and* a PR, and I realized I had not created an issue here, and I thought this might be a non-intrusive way to bring this to your attention if you had missed the PR notification over Christmas. I apologize that I misread the situation and that creating the issue it has inconvenienced you, that was not my intent.
